packages/lisk-passphrase/src/validation.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import Mnemonic from 'bip39';
+
+interface PassphraseRegularExpression {
+	readonly uppercaseRegExp: RegExp;
+	readonly whitespaceRegExp: RegExp;
+}
+interface PassphraseError {
+	readonly actual: number | boolean | string;
+	readonly code: string;
+	readonly expected: number | boolean | string;
+	readonly location?: ReadonlyArray<number>;
+	readonly message: string;
+}
+
+type RegExpReturn = RegExpMatchArray | null;
+
+const passphraseRegularExpression: PassphraseRegularExpression = {
+	uppercaseRegExp: /[A-Z]/g,
+	whitespaceRegExp: /\s/g,
+};
+
+export const countPassphraseWhitespaces = (passphrase: string): number => {
+	const whitespaceMatches: RegExpReturn = passphrase.match(
+		passphraseRegularExpression.whitespaceRegExp,
+	);
+
+	return whitespaceMatches !== null ? whitespaceMatches.length : 0;
+};
+
+export const countPassphraseWords = (passphrase: string): number =>
+	passphrase.split(' ').filter(Boolean).length;
+
+export const countUppercaseCharacters = (passphrase: string): number => {
+	const uppercaseCharacterMatches: RegExpReturn = passphrase.match(
+		passphraseRegularExpression.uppercaseRegExp,
+	);
+
+	return uppercaseCharacterMatches !== null
+		? uppercaseCharacterMatches.length
+		: 0;
+};
+
+export const locateUppercaseCharacters = (
+	passphrase: string,
+): ReadonlyArray<number> =>
+	passphrase
+		.split('')
+		.reduce(
+			(
+				passphraseArray: ReadonlyArray<number>,
```

Comment:
I should've spotted earlier, but here also `passphrase Array` would not be meaningful.
I think it's better to name `upperCaseIndexes`?